helix-view/src/document.rs

```diff
@@ -1141,6 +1142,29 @@ impl Document {
         self.inlay_hints.remove(&view_id);
     }
 
+    /// Returns the id of a view that this doc contains a selection for,
+    /// making sure it is synced with the current changes
+    /// if possible or there are no selections returns current_view
+    /// otherwise uses an arbitrary view
+    pub fn get_synced_view_id(&mut self, current_view: ViewId, tree: &mut Tree) -> ViewId {
```

Comment:
I think this should be a function on editor that takes a `DocumentId` as argument.